src/main/java/mate/academy/spring/controller/MovieSessionController.java

```diff
@@ -0,0 +1,59 @@
+package mate.academy.spring.controller;
+
+import java.time.LocalDate;
+import java.util.List;
+import java.util.stream.Collectors;
+import mate.academy.spring.model.MovieSession;
+import mate.academy.spring.model.dto.MovieSessionRequestDto;
+import mate.academy.spring.model.dto.MovieSessionResponseDto;
+import mate.academy.spring.service.MovieSessionService;
+import mate.academy.spring.service.mapper.MovieSessionMapper;
+import org.springframework.format.annotation.DateTimeFormat;
+import org.springframework.web.bind.annotation.DeleteMapping;
+import org.springframework.web.bind.annotation.GetMapping;
+import org.springframework.web.bind.annotation.PathVariable;
+import org.springframework.web.bind.annotation.PostMapping;
+import org.springframework.web.bind.annotation.PutMapping;
+import org.springframework.web.bind.annotation.RequestBody;
+import org.springframework.web.bind.annotation.RequestMapping;
+import org.springframework.web.bind.annotation.RequestParam;
+import org.springframework.web.bind.annotation.RestController;
+
+@RestController
+@RequestMapping("/movie-sessions")
+public class MovieSessionController {
+    private final MovieSessionMapper mapper;
+    private final MovieSessionService service;
+
+    public MovieSessionController(MovieSessionMapper mapper, MovieSessionService service) {
+        this.mapper = mapper;
+        this.service = service;
+    }
+
+    @PostMapping
+    public MovieSessionResponseDto create(@RequestBody MovieSessionRequestDto requestDto) {
+        return mapper.toDto(service.add(mapper.toModel(requestDto)));
+    }
+
+    @GetMapping("/available")
+    public List<MovieSessionResponseDto> getAvailable(@RequestParam Long movieId,
```

Comment:
```suggestion
    public List<MovieSessionResponseDto> findAvailableSessions(@RequestParam Long movieId,
```
keep to consistency with service layer method